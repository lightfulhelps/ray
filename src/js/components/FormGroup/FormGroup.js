// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.Node,
  className?: string,
  isCheck?: boolean,
  tag?: string,
};

const FormGroup = ({ children, className, isCheck, tag: Tag = 'div', ...other }: Props) => {
  const classes = classNames(className, isCheck ? 'form-check' : 'form-group');

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default FormGroup;
