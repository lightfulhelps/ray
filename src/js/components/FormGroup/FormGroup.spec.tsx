import React from "react";
import { shallow } from "enzyme";
import merge from "lodash/merge";
import FormGroup from "./FormGroup";

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<FormGroup {...props} />);

  return { wrapper, props };
};

describe('<FormGroup />', () => {
  it('should render', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle children', () => {
    const {
      wrapper
    } = setup({ children: 'Children' });

    expect(wrapper.text()).toBe('Children');
  });

  it('should handle className', () => {
    const {
      wrapper
    } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('form-group')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle isCheck', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.hasClass('form-group')).toBe(true);
    expect(wrapper.hasClass('form-check')).toBe(false);

    wrapper.setProps({ isCheck: true });

    expect(wrapper.hasClass('form-group')).toBe(false);
    expect(wrapper.hasClass('form-check')).toBe(true);
  });

  it('should handle tag', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.type()).toBe('div');

    wrapper.setProps({ tag: 'li' });

    expect(wrapper.type()).toBe('li');
  });

  it('should pass through other props', () => {
    const {
      wrapper
    } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});