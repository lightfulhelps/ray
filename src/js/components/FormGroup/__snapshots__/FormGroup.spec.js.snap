// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> should render 1`] = `
<div
  className="form-group"
/>
`;
